appcelerator/titanium_mobile
# [TIMOB-9366] Android: Added "radial" gradient support

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-9366

**Notes:**
- This feature is intended as an alternative for devs who are drawing circles/rings via a view's `borderRadius` property, which has known issues on Google's end.
- Supports all gradient properties except `endPoint`, which on iOS is used to smear the circular gradient from the start to end point. Android does not support this.
- Updated API docs to indicate Android's support and to better clarify the offset range for linear and radial gradients.
- The unit test written for PR #9819 covers gradient testing for both Android and iOS.

**Test:**
1. Build and run "GradientRadialTest.js" attached to [TIMOB-9366](https://jira.appcelerator.org/browse/TIMOB-9366) on Android.
2. On startup, verify that you see a circle where the outer edge is blue and the center is red.
3. Drag the "Start Radius" slider to the middle.
4. Verify that the circle becomes a ring. _(ie: The center is transparent)_
5. Tap the "Back Fill End" switch to enable it.
6. Verify that the area outside of the circle is now blue.
7. Tap the "Back Fill Start" switch to enable it.
8. Verify that the center of the circle is now solid red. _(ie: No longer transparent.)_
9. Tap the "Back Fill End" switch to disable it.
10. Drag the "End Radius" slider to the left, lower than the start radius position. _(ie: You want the end radius to be less than the start radius.)_
11. Verify that the inner part of the circle is now blue with a transparent center and the part outside of the circle is solid red.
12. Touch-drag the circle with your finger and verify that the center of the circle moves with your finger.
